solidity/test/token_stake/TestTokenStake.js

```diff
@@ -177,12 +181,13 @@ contract('TokenStaking', function(accounts) {
   })
 
   it("should not allow to recover stake before undelegation period is over", async () => {
-    await delegate(operatorOne, stakingAmount);
-
-    await mineBlocks(initializationPeriod);
-    await stakingContract.undelegate(operatorOne, {from: ownerOne});
+    let tx = await delegate(operatorOne, stakingAmount)
+    let createdAt = (await web3.eth.getBlock(tx.receipt.blockNumber)).timestamp
 
-    await mineBlocks(undelegationPeriod - 1);
+    await increaseTimeTo(createdAt + initializationPeriod + 1)
+    tx = await stakingContract.undelegate(operatorOne, {from: ownerOne});
+    let undelegatedAt = await web3.eth.getBlock(tx.receipt.blockNumber);
+    await increaseTimeTo(undelegatedAt.timestamp + undelegationPeriod);
 
```

Comment:
```
  1) Contract: TokenStaking
       should not allow to recover stake before undelegation period is over:
     AssertionError: Expected throw not received
      at _callee$ (test/helpers/expectThrowWithMessage.js:9:10)
      at tryCatch (node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:303:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:117:21)
      at step (test/helpers/expectThrowWithMessage.js:7:191)
      at /Users/piotr/go/src/github.com/keep-network/keep-core/solidity/test/helpers/expectThrowWithMessage.js:7:361
      at processTicksAndRejections (internal/process/task_queues.js:86:5)

```

locally :(

Reply:
yeah try to run it again and you'll get different results 😅 I think we have to keep +1 / -1 seconds adjustments, maybe even switch to minutes?

Reply:
Yeah... To do not lose any clarity about `-1 / +1` I think we should introduce a `const timeRoundMargin` set to one minute or so and subtract this margin from all tests stressing *before* cases. For example, *should not permit undelegating at the time before initialization period passed* would switch from `await increaseTimeTo(createdAt + initializationPeriod)` to `await increaseTimeTo(createdAt + initializationPeriod - timeRoundMargin)`. All *after* cases should stay as-is. This way it's clear `createdAt + initializationPeriod` is the last time when the undelegation is allowed but we need to use `timeRoundMargin` to avoid randomly failing tests. We also need to document `timeRoundMargin` well. What do you think?

Reply:
>We also need to document timeRoundMargin well. What do you think

yep `timeRoundMargin` constant is a good idea 👌I'll update